app/scripts/controllers/modals/editPvcModal.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('EditPvcModalController', function(APIService, DataService, $filter, LimitRangesService, QuotaService, $scope, $uibModalInstance) {
+
+    var limitRangesVersion = APIService.getPreferredVersion('limitranges');
+    var resourceQuotasVersion = APIService.getPreferredVersion('resourcequotas');
+    var appliedClusterResourceQuotasVersion = APIService.getPreferredVersion('appliedclusterresourcequotas');
+
+    var amountAndUnit = $filter('amountAndUnit');
+    var usageWithUnits = $filter('usageWithUnits');
+    var usageValue = $filter('usageValue');
+
+    var allocatedAmountAndUnit = amountAndUnit($scope.pvc.spec.resources.requests['storage']);
+
+    $scope.projectName = $scope.pvc.metadata.namespace;
+    $scope.typeDisplayName = $filter('humanizeKind')($scope.pvc.metadata.name);
+    $scope.claim = {}
+    $scope.claim.capacity = Number(allocatedAmountAndUnit[0]);
+    $scope.claim.unit = allocatedAmountAndUnit[1];
+    $scope.disableButton = true;
+
+    $scope.currentCapacityUnits = angular.copy($scope.claim);
+
+    $scope.units = [{
+      value: "Mi",
+      label: "MiB",
+      type: "binary"
+    }, {
+      value: "Gi",
+      label: "GiB",
+      type: "binary"
+    }, {
+      value: "Ti",
+      label: "TiB",
+      type: "binary"
+    }, {
+      value: "M",
+      label: "MB",
+      type: "decimal"
+    }, {
+      value: "G",
+      label: "GB",
+      type: "decimal"
+    }, {
+      value: "T",
+      label: "TB",
+      type: "decimal"
+    }];
+
+    $scope.expand = function() {
+      $scope.updatedCapacity = $scope.claim.capacity + $scope.claim.unit;
+      $uibModalInstance.close($scope.updatedCapacity);
+    };
+
+    $scope.cancel = function() {
+      $uibModalInstance.dismiss('cancel');
+    };
+
+    $scope.groupUnits = function(unit) {
```

Comment:
nit: but this function doesn't actually group, I would assume "grouping" would mean returning new grouped arrays.  Thoughts?

Reply:
I agree it's not returning a grouped array but on other side it's responsible for showing the GroupWise distribution of units in the view. So that's why the name is groupUnits name I have used.

If you have another suggestion I am ok changing the name.

Reply:
Right, I think the html clarifies this:

```html
<ui-select-choices 
   repeat="option.value as option in units" 
   group-by="groupUnits">
```
The `group-by` already tells me it is grouping.  So the function you pass can be named something more specifc, like `unitTextLabel()` for example.   `group-by="unitTextLabel"` gives me a better idea of what the grouping actually is.

Reply:
ack!